### DISCUSSION TITLE
Can (non-Boot) Java config be done 2.3.x?

### DISCUSSION THREAD

**Discussion by jbisotti** [awspring/spring-cloud-aws#203] [Q&A] [answered]
In the 2.3.0 Release Notes, it is mentioned that XML config is deprecated. Is there non-Boot Java config equivalent/replacement? Or is it deprecated so it can be completely replaced in 3.x with Java config?

I've read the latest Reference doc and Googled quite a bit. It's possible I just missed it but nothing jumped out at me.

I'm reading/writing from/to S3 so I'm specifically interested in replacements for `<aws-context:context-credentials>` & `<aws-context:context-resource-loader>`.

Thanks.

**Comment by maciejwalkowiak** [comment, marked as answer]
We have deprecated both XML configuration and `@EnableX` annotations simply because we don't have time to support all possible configuration options. The replacement for them in non-boot application is do do manually what they do - create `CredentialsProvider` bean and related resources for S3 (see `ContextResourceLoaderConfiguration`).

**Reply by jbisotti** [reply]
Thank you for the answer.
